### PR TITLE
Build: Add --dry-run

### PR DESCRIPTION
**Pull request by gibson042**
And improve the overall development experience.

### CODE REVIEW

File: release.js [hunk 1]
```diff
@@ -15,8 +15,10 @@ var commonTasks, stableTasks,
 		}
 	};
 
-fs.readdirSync( "./lib" ).forEach(function( module ) {
-	require( "./lib/" + module )( Release );
+fs.readdirSync( "./lib" ).forEach(function( filename ) {
+	if ( filename.slice( -3 ) === ".js" ) {
```
Comment (mgol) [inline release.js]: `filename.endsWith( ".js" )` would be clearer.

File: README.md [hunk 2]
```diff
@@ -40,14 +40,14 @@ cd $cdn
 git push -f
 cd -
 
-rm -rf __release/
-node release.js --remote=$project
+npm run clean
+env TEST_REMOTE=$cdn node --dry-run release.js --remote=$project
```
[the remark this answers is not in the document]
Comment (gibson042) [reply README.md]: The script will try to clone git@github.com:jquery/fake-cdn.git, and will fail unless the user has access.

File: release.js (hunk 1 above)
Comment (gibson042) [reply release.js]: Agreed; updated.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgol) [reply README.md]: I meant skipping just `env `, not `env TEST_REMOTE=$cdn `. I've always just written:
```
ENV1=VALUE1 ENV2=VALUE2 command
```
I was wondering what difference does prefixing with `env ` actually make.

[the remark this answers is not in the document]
Comment (timmywil) [reply README.md]: I agree I don't often see the use of `env`, but I'm not opposed.

**Pull request merged**